`db-migrate db:create` not working as intended

I've been playing around for most of the morning with db-migrate and getting myself familiar with the ins and outs. One thing I have noticed is that the db:create command doesn't seem to work as I would have expected it. 

For example in my `database.json` I have configuration for a test database and a dev database. Now, when I run the `db-migrate db:create myapptest` I would expect to see that it creates a database to run my tests against. instead I get the following error: 

`[ERROR] Error: ER_BAD_DB_ERROR: Unknown database 'myapptest'` 

the whole reason for running the `db:create` command is _because_ that database doesn't exist and I want to create it.

Yet if I create the database and then run the command passing a a random name it will happily create any database I want. It will also claim that I have created the same database multiple times even when it has in fact created it once. 

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/43274257-db-migrate-db-create-not-working-as-intended?utm_campaign=plugin&utm_content=tracker%2F73887&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F73887&utm_medium=issues&utm_source=github).
</bountysource-plugin>

This occurs when trying to use Postgres as well. My driver is "pg", and I've installed `db-migrate-pg`.

```
>  db-migrate db:create $( jq ".local.database" database.json) -e local[ERROR] AssertionError [ERR_ASSERTION]: ifError got unwanted exception: database "aegisdb" does not exist
    at /usr/local/lib/node_modules/db-migrate/lib/commands/db.js:17:12
    at /usr/local/lib/node_modules/db-migrate/lib/driver/index.js:95:9
    at /usr/local/lib/node_modules/db-migrate-pg/index.js:662:7
```

On db-migrate 0.11.3, btw.

--- a/lib/commands/db.js
+++ b/lib/commands/db.js
@@ -13,6 +13,8 @@ function executeDB (internals, config, callback) {
     return;
   }
 
+  delete config.getCurrent().settings.database;
+
   index.driver(config.getCurrent().settings, function (err, db) {
     assert.ifError(err);
 

